python/whylogs/migration/converters.py

```diff
@@ -84,19 +180,131 @@ def _extract_type_counts_metric(msg: ColumnMessageV0) -> TypeCountersMetric:
     )
 
 
+def _extract_schema_message_v0(col_prof: ColumnProfileView) -> SchemaMessageV0:
+    metric: TypeCountersMetric = col_prof.get_metric(TypeCountersMetric.get_namespace())
+    if metric is None:
+        return SchemaMessageV0()
+
+    type_counts: Dict[int, int] = {}
+    type_counts[InferredType.INTEGRAL] = metric.integral.value
+    type_counts[InferredType.BOOLEAN] = metric.boolean.value
+    type_counts[InferredType.FRACTIONAL] = metric.fractional.value
+    type_counts[InferredType.STRING] = metric.string.value
+    type_counts[InferredType.UNKNOWN] = metric.object.value
+
+    msg_v0 = SchemaMessageV0(
+        typeCounts=type_counts,
+    )
+
+    return msg_v0
+
+
 def _extract_col_counts(msg: ColumnMessageV0) -> ColumnCountsMetric:
     count_n = msg.counters.count
     count_null = msg.counters.null_count
     return ColumnCountsMetric(n=IntegralComponent(count_n or 0), null=IntegralComponent(count_null.value or 0))
 
 
+def _extract_counters_v0(col_prof: ColumnProfileView) -> CountersV0:
+    counts_metric: ColumnCountsMetric = col_prof.get_metric(ColumnCountsMetric.get_namespace())
+    count_options = dict(count=0)
+
+    if counts_metric is None:
+        return CountersV0(**dict(count=0))
+
+    count_options = dict(count=counts_metric.n.value)
+
+    return CountersV0(**count_options)
+
+
 def _extract_dist_metric(msg: ColumnMessageV0) -> DistributionMetric:
-    floats_sk = ds.kll_floats_sketch.deserialize(msg.numbers.histogram)
-    doubles_sk: ds.kll_doubles_sketch = ds.kll_floats_sketch.float_to_doubles(floats_sk)
+    kll_bytes = msg.numbers.histogram
+    floats_sk = None
+    doubles_sk: Optional[ds.kll_doubles_sketch] = None
+    try:
+        floats_sk = ds.kll_floats_sketch.deserialize(kll_bytes)
+    except ValueError as e:
+        logger.info(f"kll encountered old format which threw exception: {e}, attempting kll_doubles deserialization.")
+    if floats_sk is None:
+        doubles_sk = ds.kll_doubles_sketch.deserialize(kll_bytes)
+    else:
+        doubles_sk = ds.kll_floats_sketch.float_to_doubles(floats_sk)
+
     dist_mean = msg.numbers.variance.mean
     dist_m2 = msg.numbers.variance.sum
     return DistributionMetric(
         kll=KllComponent(doubles_sk),
         mean=FractionalComponent(dist_mean),
         m2=FractionalComponent(dist_m2),
     )
+
+
+def _extract_frequent_items_sketch_message_v0(col_prof: ColumnProfileView) -> FrequentItemsSketchMessageV0:
+    frequent_items_metric: FrequentItemsMetric = col_prof.get_metric(FrequentItemsMetric.get_namespace())
+    frequent_items_v0 = FrequentItemsSketchMessageV0(sketch=EMPTY_FI_SKETCH)
+    if frequent_items_metric is None:
+        return frequent_items_v0
+
+    frequent_items_v0.sketch = frequent_items_metric.frequent_strings.value.serialize()
+    frequent_items_v0.lg_max_k = 12
```

Comment:
Why 12?

Reply:
it was the v0 default, I can create constants for these

Reply:
constants created for all default values.